util/chplenv/third_party_utils.py

```diff
@@ -62,70 +60,152 @@ def handle_la(la_path):
     return args
 
 #
-# Return compiler arguments required to use a library known to
-# pkgconfig. The pkg can be a path to a .pc file or the name of a
-# system-installed package or the name of a third-party package.
+# Return compiler arguments required to use a system library known to
+# pkg-config. The pkg argument should be the name of a system-installed
+# package.
 #
-# if system=True, searches for a system-installed package.
+# Returns a 2-tuple of lists
+#  (compiler_bundled_args, compiler_system_args)
 @memoize
-def pkgconfig_get_compile_args(pkg, ucp='', system=True):
-  havePcFile = pkg.endswith('.pc')
-  pcArg = pkg
-  if not havePcFile:
-    if system:
-      # check that pkg-config knows about the package in question
-      run_command(['pkg-config', '--exists', pkg])
-    else:
-      # look for a .pc file
-      if ucp == '':
-        ucp = default_uniq_cfg_path()
-      pcfile = pkg + '.pc' # maybe needs to be an argument later?
+def pkgconfig_get_system_compile_args(pkg):
+    # check that pkg-config knows about the package in question
+    run_command(['pkg-config', '--exists', pkg])
+    # run pkg-config to get the cflags
+    cflags_line = run_command(['pkg-config', '--cflags'] + [pkg]);
+    cflags = cflags_line.split()
+    return ([ ], cflags)
+
+#
+# Return compiler arguments required to use a bundled library
+# based on a pkg-config .pc file.
+#
+# Even though pkg-config is not available by default on Mac OS X,
+# many third-party packages create .pc files. This function can
+# parse them. Additionally, it adjusts the output to replace
+# paths to a previous third-party install directory with the current one.
+#
+# pkg is the name of the subdir in third-party
+# ucp is the unique config path
+# pcfile is the name of the .pc file
+#
+# This function looks for
+#   third-party/<pkg>/install/<ucp>/lib/pkgconfig/<pcfile>
+#
+# Returns a 2-tuple of lists
+#  (compiler_bundled_args, compiler_system_args)
+def pkgconfig_get_bundled_compile_args(pkg, ucp, pcfile):
+    install_path = get_bundled_install_path(pkg, ucp)
+
+    # give up early if the 3rd party package hasn't been built
+    if not os.path.exists(install_path):
+        return ([ ], [ ])
+
+    pcpath = os.path.join(install_path, 'lib', 'pkgconfig', pcfile)
+
+    # if we get this far, we should have a .pc file. check that it exists.
+    if not os.access(pcpath, os.R_OK):
+        error("Could not find '{0}'".format(pcpath), ValueError)
+
+    d = read_pkg_config_file(pcpath,
+                             os.path.join('third-party', pkg, 'install', ucp),
+                             install_path)
 
-      pcArg = os.path.join(get_cfg_install_path(pkg, ucp), 'lib',
-                           'pkgconfig', pcfile)
+    if 'Requires' in d and d['Requires']:
+        warning("Simple pkg-config parser does not handle Requires")
+        warning("in {0}".format(pcpath))
```

Comment:
This will print out as:

```
Warning: Simple pkg-config parser does not handle Requires
Warning: in <path>.
```

Is that what you want, or do you just want a single warning with line/break?

Reply:
That's what I wanted but I'm open to suggestions. Do you like it better without the second Warning?

Reply:
No, it's fine with me. Just making sure that's the formatting you wanted